server/src/main/java/org/apache/druid/metadata/SQLMetadataConnector.java

```diff
@@ -587,6 +587,11 @@ protected void alterSegmentTable()
     Map<String, String> columnNameTypes = new HashMap<>();
     columnNameTypes.put("used_status_last_updated", "VARCHAR(255)");
 
+    // upgraded_from_segment_id is the first segment to which the same load spec originally belonged
+    // Load specs can be shared as a result of segment version upgrade
+    // This column is null for segments that haven't been upgraded.
+    columnNameTypes.put("upgraded_from_segment_id", "VARCHAR(255)");
```

Comment:
For completeness, should we also add a check in `validateSegmentsTable()` to look for the presence of the new column `upgraded_from_segment_id`? When auto-table creation isn't enabled, we should let the operator know that the column and index should be added manually if it's not already there.

Reply:
I think this is an existing shortcoming that applies to other tables too.
Maybe we should deal with it in a separate PR.